File: tasks/webServer-golang/webServer.go
package main

import "fmt"
import "net"
import "log"
import "bufio"
import "os"
import "strings"


func main(){
	ln, err := net.Listen("tcp", ":8081")
if err != nil {

	fmt.Println(err)
}

for {
	conn, err := ln.Accept()
	if err != nil {
		log.Print(err)
	}

	go handler(conn)

}

}


func removeClient(conn net.Conn) {
	conn.Close()
}

func handler(conn net.Conn) {

	errorChan := make(chan error)
	dataChan := make(chan []byte)

	go readWrapper(conn, dataChan, errorChan)
	data := <-dataChan
	res1 := strings.Split(string(data), "\n")
	_ = res1
	// b := []byte("HTTP/1.1 200 OK Hello, World!")
	conn.Write([]byte("HTTP/1.0 200 OK\r\nContent-Type:image/png\r\n\r\n"))
	go ReadFile(conn)




}

func readWrapper(conn net.Conn, dataChan chan []byte, errorChan chan error) {
	for {
		buf := make([]byte, 65538)
		reqLen, err := conn.Read(buf)
		if err != nil {
			errorChan <- err
			return
		}
		dataChan <- buf[:reqLen]
	}

}


func ReadFile(conn net.Conn) {
	defer removeClient(conn)


    f, err := os.Open("discord-0.0.12.deb")

    if err != nil {
        log.Fatal(err)
    }


    reader := bufio.NewReader(f)
    buf := make([]byte, 65538)

    for {

		n, err := reader.Read(buf)

        if err != nil {
            break
        }
		conn.Write(buf[:n])


	}

}

File: tasks/webServer-Deno/webServer.js
const encoder = new TextEncoder();
const decoder = new TextDecoder();

const listener = Deno.listen({ port: 8080 });

console.log("listening on 0.0.0.0:8080");
for await (const conn of listener) {
  const buf = new Uint8Array(1024);
  await conn.read(buf);
  await conn.write(encoder.encode('HTTP/1.0 200 OK\r\n\r\n'))
  const file = await Deno.open("./discord-0.0.12.deb");
  await Deno.copy(file, conn);
  file.close();
  conn.close();
}

// for await (const conn of listener) {
//   const buf = new Uint8Array(1024);
//   await conn.read(buf);
//   await conn.write(encoder.encode('HTTP/1.0 200 OK\r\nContent-Type:image/png\r\n\r\n'))
//   const file = await Deno.open("./peng.png");
//   await Deno.copy(file, conn);
//   file.close();
//   conn.close();
// }
